src/sequoia-chameleon-gnupg/debian/gpg-sq.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH gpg-sq 1  "gpg-sq 0.5.1" 
.SH NAME
gpg\-sq \- This is a re\-implementation and drop\-in replacement of gpg using the Sequoia OpenPGP implementation.

gpg\-sq is not feature\-complete. It currently implements a commonly used subset of the signature creation and verification commands, the encryption and decryption commands, the key listing commands, and some miscellaneous commands.

Support for trust models is limited. Currently, the Web\-of\-Trust ("pgp") and always trust ("always") are implemented.
.SH SYNOPSIS
\fBgpg\-sq\fR [\fB\-s\fR|\fB\-\-sign\fR] [\fB\-\-clear\-sign\fR] [\fB\-\-clearsign\fR] [\fB\-b\fR|\fB\-\-detach\-sign\fR] [\fB\-e\fR|\fB\-\-encrypt\fR] [\fB\-\-encrypt\-files\fR] [\fB\-c\fR|\fB\-\-symmetric\fR] [\fB\-\-store\fR] [\fB\-d\fR|\fB\-\-decrypt\fR] [\fB\-\-decrypt\-files\fR] [\fB\-\-verify\fR] [\fB\-\-verify\-files\fR] [\fB\-k\fR|\fB\-\-list\-keys\fR] [\fB\-\-list\-public\-keys\fR] [\fB\-\-list\-signatures\fR] [\fB\-\-list\-sigs\fR] [\fB\-\-check\-signatures\fR] [\fB\-\-check\-sigs\fR] [\fB\-\-fingerprint\fR] [\fB\-K\fR|\fB\-\-list\-secret\-keys\fR] [\fB\-\-generate\-key\fR] [\fB\-\-gen\-key\fR] [\fB\-\-quick\-generate\-key\fR] [\fB\-\-quick\-gen\-key\fR] [\fB\-\-quick\-add\-uid\fR] [\fB\-\-quick\-adduid\fR] [\fB\-\-quick\-add\-key\fR] [\fB\-\-quick\-addkey\fR] [\fB\-\-quick\-revoke\-uid\fR] [\fB\-\-quick\-revuid\fR] [\fB\-\-quick\-set\-expire\fR] [\fB\-\-quick\-set\-primary\-uid\fR] [\fB\-\-full\-generate\-key\fR] [\fB\-\-full\-gen\-key\fR] [\fB\-\-generate\-revocation\fR] [\fB\-\-gen\-revoke\fR] [\fB\-\-delete\-keys\fR] [\fB\-\-delete\-secret\-keys\fR] [\fB\-\-quick\-sign\-key\fR] [\fB\-\-quick\-lsign\-key\fR] [\fB\-\-quick\-revoke\-sig\fR] [\fB\-\-sign\-key\fR] [\fB\-\-lsign\-key\fR] [\fB\-\-edit\-key\fR] [\fB\-\-key\-edit\fR] [\fB\-\-change\-passphrase\fR] [\fB\-\-passwd\fR] [\fB\-\-generate\-designated\-revocation\fR] [\fB\-\-desig\-revoke\fR] [\fB\-\-export\fR] [\fB\-\-send\-keys\fR] [\fB\-\-receive\-keys\fR] [\fB\-\-recv\-keys\fR] [\fB\-\-search\-keys\fR] [\fB\-\-refresh\-keys\fR] [\fB\-\-locate\-keys\fR] [\fB\-\-locate\-external\-keys\fR] [\fB\-\-fetch\-keys\fR] [\fB\-\-show\-keys\fR] [\fB\-\-export\-secret\-keys\fR] [\fB\-\-export\-secret\-subkeys\fR] [\fB\-\-export\-ssh\-key\fR] [\fB\-\-import\fR] [\fB\-\-fast\-import\fR] [\fB\-\-list\-config\fR] [\fB\-\-list\-gcrypt\-config\fR] [\fB\-\-gpgconf\-list\fR] [\fB\-\-gpgconf\-test\fR] [\fB\-\-list\-packets\fR] [\fB\-\-export\-ownertrust\fR] [\fB\-\-import\-ownertrust\fR] [\fB\-\-update\-trustdb\fR] [\fB\-\-check\-trustdb\fR] [\fB\-\-fix\-trustdb\fR] [\fB\-\-list\-trustdb\fR] [\fB\-\-dearmor\fR] [\fB\-\-dearmour\fR] [\fB\-\-enarmor\fR] [\fB\-\-enarmour\fR] [\fB\-\-print\-md\fR] [\fB\-\-print\-mds\fR] [\fB\-\-gen\-prime\fR] [\fB\-\-gen\-random\fR] [\fB\-\-server\fR] [\fB\-\-tofu\-policy\fR] [\fB\-\-delete\-secret\-and\-public\-keys\fR] [\fB\-\-rebuild\-keydb\-caches\fR] [\fB\-\-list\-key\fR] [\fB\-\-list\-sig\fR] [\fB\-\-check\-sig\fR] [\fB\-\-show\-key\fR] [\fB\-\-Monitor\fR] [\fB\-v\fR|\fB\-\-verbose\fR] [\fB\-\-no\-verbose\fR] [\fB\-q\fR|\fB\-\-quiet\fR] [\fB\-\-no\-tty\fR] [\fB\-\-no\-greeting\fR] [\fB\-\-debug\fR] [\fB\-\-debug\-level\fR] [\fB\-\-debug\-all\fR] [\fB\-\-debug\-iolbf\fR] [\fB\-\-display\-charset\fR] [\fB\-\-charset\fR] [\fB\-\-options\fR] [\fB\-\-no\-options\fR] [\fB\-\-logger\-fd\fR] [\fB\-\-log\-file\fR] [\fB\-\-logger\-file\fR] [\fB\-\-debug\-quick\-random\fR] [\fB\-\-Configuration\fR] [\fB\-\-homedir\fR] [\fB\-\-faked\-system\-time\fR] [\fB\-\-default\-key\fR] [\fB\-\-encrypt\-to\fR] [\fB\-\-no\-encrypt\-to\fR] [\fB\-\-hidden\-encrypt\-to\fR] [\fB\-\-encrypt\-to\-default\-key\fR] [\fB\-\-default\-recipient\fR] [\fB\-\-default\-recipient\-self\fR] [\fB\-\-no\-default\-recipient\fR] [\fB\-\-group\fR] [\fB\-\-ungroup\fR] [\fB\-\-no\-groups\fR] [\fB\-\-compliance\fR] [\fB\-\-gnupg\fR] [\fB\-\-no\-pgp2\fR] [\fB\-\-no\-pgp6\fR] [\fB\-\-no\-pgp7\fR] [\fB\-\-no\-pgp8\fR] [\fB\-\-rfc2440\fR] [\fB\-\-rfc4880\fR] [\fB\-\-rfc4880bis\fR] [\fB\-\-openpgp\fR] [\fB\-\-pgp6\fR] [\fB\-\-pgp7\fR] [\fB\-\-pgp8\fR] [\fB\-\-default\-new\-key\-algo\fR] [\fB\-\-min\-rsa\-length\fR] [\fB\-\-always\-trust\fR] [\fB\-\-trust\-model\fR] [\fB\-\-photo\-viewer\fR] [\fB\-\-known\-notation\fR] [\fB\-\-agent\-program\fR] [\fB\-\-dirmngr\-program\fR] [\fB\-\-exit\-on\-status\-write\-error\fR] [\fB\-\-limit\-card\-insert\-tries\fR] [\fB\-\-enable\-progress\-filter\fR] [\fB\-\-temp\-directory\fR] [\fB\-\-exec\-path\fR] [\fB\-\-expert\fR] [\fB\-\-no\-expert\fR] [\fB\-\-no\-secmem\-warning\fR] [\fB\-\-require\-secmem\fR] [\fB\-\-no\-require\-secmem\fR] [\fB\-\-no\-permission\-warning\fR] [\fB\-n\fR|\fB\-\-dry\-run\fR] [\fB\-i\fR|\fB\-\-interactive\fR] [\fB\-\-default\-sig\-expire\fR] [\fB\-\-ask\-sig\-expire\fR] [\fB\-\-no\-ask\-sig\-expire\fR] [\fB\-\-default\-cert\-expire\fR] [\fB\-\-ask\-cert\-expire\fR] [\fB\-\-no\-ask\-cert\-expire\fR] [\fB\-\-default\-cert\-level\fR] [\fB\-\-min\-cert\-level\fR] [\fB\-\-ask\-cert\-level\fR] [\fB\-\-no\-ask\-cert\-level\fR] [\fB\-\-only\-sign\-text\-ids\fR] [\fB\-\-enable\-large\-rsa\fR] [\fB\-\-disable\-large\-rsa\fR] [\fB\-\-enable\-dsa2\fR] [\fB\-\-disable\-dsa2\fR] [\fB\-\-personal\-cipher\-preferences\fR] [\fB\-\-personal\-digest\-preferences\fR] [\fB\-\-personal\-compress\-preferences\fR] [\fB\-\-default\-preference\-list\fR] [\fB\-\-default\-keyserver\-url\fR] [\fB\-\-no\-expensive\-trust\-checks\fR] [\fB\-\-allow\-non\-selfsigned\-uid\fR] [\fB\-\-no\-allow\-non\-selfsigned\-uid\fR] [\fB\-\-allow\-freeform\-uid\fR] [\fB\-\-no\-allow\-freeform\-uid\fR] [\fB\-\-preserve\-permissions\fR] [\fB\-\-default\-cert\-check\-level\fR] [\fB\-\-tofu\-default\-policy\fR] [\fB\-\-lock\-once\fR] [\fB\-\-lock\-multiple\fR] [\fB\-\-lock\-never\fR] [\fB\-\-compress\-algo\fR] [\fB\-\-compression\-algo\fR] [\fB\-\-bzip2\-decompress\-lowmem\fR] [\fB\-\-completes\-needed\fR] [\fB\-\-marginals\-needed\fR] [\fB\-\-max\-cert\-depth\fR] [\fB\-\-trustdb\-name\fR] [\fB\-\-auto\-check\-trustdb\fR] [\fB\-\-no\-auto\-check\-trustdb\fR] [\fB\-\-force\-ownertrust\fR] [\fB\-\-Input\fR] [\fB\-\-multifile\fR] [\fB\-\-input\-size\-hint\fR] [\fB\-\-utf8\-strings\fR] [\fB\-\-no\-utf8\-strings\fR] [\fB\-\-set\-filesize\fR] [\fB\-\-no\-literal\fR] [\fB\-\-set\-notation\fR] [\fB\-\-sig\-notation\fR] [\fB\-\-cert\-notation\fR] [\fB\-\-set\-policy\-url\fR] [\fB\-\-sig\-policy\-url\fR] [\fB\-\-cert\-policy\-url\fR] [\fB\-\-sig\-keyserver\-url\fR] [\fB\-\-Output\fR] [\fB\-a\fR|\fB\-\-armor\fR] [\fB\-\-armour\fR] [\fB\-\-no\-armor\fR] [\fB\-\-no\-armour\fR] [\fB\-o\fR|\fB\-\-output\fR] [\fB\-\-max\-output\fR] [\fB\-\-comment\fR] [\fB\-\-default\-comment\fR] [\fB\-\-no\-comments\fR] [\fB\-\-emit\-version\fR] [\fB\-\-no\-emit\-version\fR] [\fB\-\-no\-version\fR] [\fB\-\-not\-dash\-escaped\fR] [\fB\-\-escape\-from\-lines\fR] [\fB\-\-no\-escape\-from\-lines\fR] [\fB\-\-mimemode\fR] [\fB\-\-textmode\fR] [\fB\-\-no\-textmode\fR] [\fB\-\-set\-filename\fR] [\fB\-\-for\-your\-eyes\-only\fR] [\fB\-\-no\-for\-your\-eyes\-only\fR] [\fB\-\-show\-notation\fR] [\fB\-\-no\-show\-notation\fR] [\fB\-\-show\-session\-key\fR] [\fB\-\-use\-embedded\-filename\fR] [\fB\-\-no\-use\-embedded\-filename\fR] [\fB\-\-unwrap\fR] [\fB\-\-mangle\-dos\-filenames\fR] [\fB\-\-no\-mangle\-dos\-filenames\fR] [\fB\-\-no\-symkey\-cache\fR] [\fB\-\-skip\-verify\fR] [\fB\-\-list\-only\fR] [\fB\-z \fR] [\fB\-\-compress\-level\fR] [\fB\-\-bzip2\-compress\-level\fR] [\fB\-\-disable\-signer\-uid\fR] [\fB\-\-ImportExport\fR] [\fB\-\-auto\-key\-locate\fR] [\fB\-\-no\-auto\-key\-locate\fR] [\fB\-\-auto\-key\-import\fR] [\fB\-\-no\-auto\-key\-import\fR] [\fB\-\-auto\-key\-retrieve\fR] [\fB\-\-no\-auto\-key\-retrieve\fR] [\fB\-\-include\-key\-block\fR] [\fB\-\-no\-include\-key\-block\fR] [\fB\-\-disable\-dirmngr\fR] [\fB\-\-keyserver\fR] [\fB\-\-keyserver\-options\fR] [\fB\-\-key\-origin\fR] [\fB\-\-import\-options\fR] [\fB\-\-import\-filter\fR] [\fB\-\-export\-options\fR] [\fB\-\-export\-filter\fR] [\fB\-\-merge\-only\fR] [\fB\-\-allow\-secret\-key\-import\fR] [\fB\-\-Keylist\fR] [\fB\-\-list\-options\fR] [\fB\-\-show\-photos\fR] [\fB\-\-no\-show\-photos\fR] [\fB\-\-show\-policy\-url\fR] [\fB\-\-no\-show\-policy\-url\fR] [\fB\-\-with\-colons\fR] [\fB\-\-with\-tofu\-info\fR] [\fB\-\-with\-key\-data\fR] [\fB\-\-with\-sig\-list\fR] [\fB\-\-with\-sig\-check\fR] [\fB\-\-with\-fingerprint\fR] [\fB\-\-with\-subkey\-fingerprint\fR] [\fB\-\-with\-subkey\-fingerprints\fR] [\fB\-\-with\-icao\-spelling\fR] [\fB\-\-with\-keygrip\fR] [\fB\-\-with\-secret\fR] [\fB\-\-with\-wkd\-hash\fR] [\fB\-\-with\-key\-origin\fR] [\fB\-\-fast\-list\-mode\fR] [\fB\-\-fixed\-list\-mode\fR] [\fB\-\-legacy\-list\-mode\fR] [\fB\-\-print\-pka\-records\fR] [\fB\-\-print\-dane\-records\fR] [\fB\-\-keyid\-format\fR] [\fB\-\-show\-keyring\fR] [\fB\-r\fR|\fB\-\-recipient\fR] [\fB\-\-hidden\-recipient\fR] [\fB\-\-recipient\-file\fR] [\fB\-\-hidden\-recipient\-file\fR] [\fB\-\-remote\-user\fR] [\fB\-\-throw\-keyids\fR] [\fB\-\-no\-throw\-keyids\fR] [\fB\-u\fR|\fB\-\-local\-user\fR] [\fB\-\-trusted\-key\fR] [\fB\-\-sender\fR] [\fB\-\-try\-secret\-key\fR] [\fB\-\-try\-all\-secrets\fR] [\fB\-\-no\-default\-keyring\fR] [\fB\-\-no\-keyring\fR] [\fB\-\-keyring\fR] [\fB\-\-primary\-keyring\fR] [\fB\-\-secret\-keyring\fR] [\fB\-\-skip\-hidden\-recipients\fR] [\fB\-\-no\-skip\-hidden\-recipients\fR] [\fB\-\-override\-session\-key\fR] [\fB\-\-override\-session\-key\-fd\fR] [\fB\-\-Security\fR] [\fB\-\-s2k\-mode\fR] [\fB\-\-s2k\-digest\-algo\fR] [\fB\-\-s2k\-cipher\-algo\fR] [\fB\-\-s2k\-count\fR] [\fB\-\-require\-backsigs\fR] [\fB\-\-require\-cross\-certification\fR] [\fB\-\-no\-require\-backsigs\fR] [\fB\-\-no\-require\-cross\-certification\fR] [\fB\-\-verify\-options\fR] [\fB\-\-enable\-special\-filenames\fR] [\fB\-\-no\-random\-seed\-file\fR] [\fB\-\-no\-sig\-cache\fR] [\fB\-\-ignore\-time\-conflict\fR] [\fB\-\-ignore\-valid\-from\fR] [\fB\-\-ignore\-crc\-error\fR] [\fB\-\-ignore\-mdc\-error\fR] [\fB\-\-disable\-cipher\-algo\fR] [\fB\-\-disable\-pubkey\-algo\fR] [\fB\-\-cipher\-algo\fR] [\fB\-\-digest\-algo\fR] [\fB\-\-cert\-digest\-algo\fR] [\fB\-\-override\-compliance\-check\fR] [\fB\-\-allow\-weak\-key\-signatures\fR] [\fB\-\-allow\-weak\-digest\-algos\fR] [\fB\-\-weak\-digest\fR] [\fB\-\-allow\-multisig\-verification\fR] [\fB\-\-allow\-multiple\-messages\fR] [\fB\-\-no\-allow\-multiple\-messages\fR] [\fB\-\-batch\fR] [\fB\-\-no\-batch\fR] [\fB\-\-yes\fR] [\fB\-\-no\fR] [\fB\-\-status\-fd\fR] [\fB\-\-status\-file\fR] [\fB\-\-attribute\-fd\fR] [\fB\-\-attribute\-file\fR] [\fB\-\-command\-fd\fR] [\fB\-\-command\-file\fR] [\fB\-\-passphrase\fR] [\fB\-\-passphrase\-fd\fR] [\fB\-\-passphrase\-file\fR] [\fB\-\-passphrase\-repeat\fR] [\fB\-\-pinentry\-mode\fR] [\fB\-\-force\-sign\-key\fR] [\fB\-\-request\-origin\fR] [\fB\-\-display\fR] [\fB\-\-ttyname\fR] [\fB\-\-ttytype\fR] [\fB\-\-lc\-ctype\fR] [\fB\-\-lc\-messages\fR] [\fB\-\-xauthority\fR] [\fB\-\-no\-autostart\fR] [\fB\-\-forbid\-gen\-key\fR] [\fB\-\-require\-compliance\fR] [\fB\-\-use\-only\-openpgp\-card\fR] [\fB\-\-rfc2440\-text\fR] [\fB\-\-no\-rfc2440\-text\fR] [\fB\-\-personal\-cipher\-prefs\fR] [\fB\-\-personal\-digest\-prefs\fR] [\fB\-\-personal\-compress\-prefs\fR] [\fB\-\-sign\-with\fR] [\fB\-\-user\fR] [\fB\-\-use\-agent\fR] [\fB\-\-no\-use\-agent\fR] [\fB\-\-gpg\-agent\-info\fR] [\fB\-\-reader\-port\fR] [\fB\-\-ctapi\-driver\fR] [\fB\-\-pcsc\-driver\fR] [\fB\-\-disable\-ccid\fR] [\fB\-\-honor\-http\-proxy\fR] [\fB\-\-tofu\-db\-format\fR] [\fB\-\-strict\fR] [\fB\-\-no\-strict\fR] [\fB\-\-load\-extension\fR] [\fB\-\-sk\-comments\fR] [\fB\-\-no\-sk\-comments\fR] [\fB\-\-compress\-keys\fR] [\fB\-\-compress\-sigs\fR] [\fB\-\-force\-v3\-sigs\fR] [\fB\-\-no\-force\-v3\-sigs\fR] [\fB\-\-force\-v4\-certs\fR] [\fB\-\-no\-force\-v4\-certs\fR] [\fB\-\-no\-mdc\-warning\fR] [\fB\-\-force\-mdc\fR] [\fB\-\-no\-force\-mdc\fR] [\fB\-\-disable\-mdc\fR] [\fB\-\-no\-disable\-mdc\fR] [\fB\-\-x\-sequoia\-parcimonie\fR] [\fB\-\-x\-sequoia\-autostart\-parcimonie\fR] [\fB\-\-x\-sequoia\-parcimonie\-daemonize\fR] [\fB\-\-warranty\fR] [\fB\-\-dump\-option\-table\fR] [\fB\-\-dump\-options\fR] [\fB\-h\fR|\fB\-\-help\fR] 
.SH DESCRIPTION
This is a re\-implementation and drop\-in replacement of gpg using the Sequoia OpenPGP implementation.
.PP
gpg\-sq is not feature\-complete. It currently implements a commonly used subset of the signature creation and verification commands, the encryption and decryption commands, the key listing commands, and some miscellaneous commands.
.PP
Support for trust models is limited. Currently, the Web\-of\-Trust ("pgp") and always trust ("always") are implemented.
.SH OPTIONS
.TP
\fB\-s\fR, \fB\-\-sign\fR
make a signature
.TP
\fB\-\-clear\-sign\fR
make a clear text signature
.TP
\fB\-\-clearsign\fR
@
.TP
\fB\-b\fR, \fB\-\-detach\-sign\fR
make a detached signature
.TP
\fB\-e\fR, \fB\-\-encrypt\fR
encrypt data
.TP
\fB\-\-encrypt\-files\fR
@
.TP
\fB\-c\fR, \fB\-\-symmetric\fR
encryption only with symmetric cipher
.TP
\fB\-\-store\fR
@
.TP
\fB\-d\fR, \fB\-\-decrypt\fR
decrypt data (default)
.TP
\fB\-\-decrypt\-files\fR
@
.TP
\fB\-\-verify\fR
verify a signature
.TP
\fB\-\-verify\-files\fR
@
.TP
\fB\-k\fR, \fB\-\-list\-keys\fR
list keys
.TP
\fB\-\-list\-public\-keys\fR
@
.TP
\fB\-\-list\-signatures\fR
list keys and signatures
.TP
\fB\-\-list\-sigs\fR
@
.TP
\fB\-\-check\-signatures\fR
list and check key signatures
.TP
\fB\-\-check\-sigs\fR
@
.TP
\fB\-\-fingerprint\fR
list keys and fingerprints
.TP
\fB\-K\fR, \fB\-\-list\-secret\-keys\fR
list secret keys
.TP
\fB\-\-generate\-key\fR
generate a new key pair
.TP
\fB\-\-gen\-key\fR
@
.TP
\fB\-\-quick\-generate\-key\fR
quickly generate a new key pair
.TP
\fB\-\-quick\-gen\-key\fR
@
.TP

NEWS.rst
2021-10-28
==========

semver_suffix is broken in debcargo 2.4.4.

There is a fix in debcargo.git master. To use it, first build it then run::

  $ DEBCARGO=$PWD/../debcargo/target/debug/debcargo ./update.sh <args>

or some variation of it, depending on your paths.


src/sequoia-chameleon-gnupg/debian/completions/gpg-sq.bash
_gpg-sq() {
    local i cur prev opts cmd
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="gpg__sq"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        gpg__sq)
            opts="-s -b -e -c -d -k -K -v -q -n -i -a -o -z -r -u -h --sign --clear-sign --clearsign --detach-sign --encrypt --encrypt-files --symmetric --store --decrypt --decrypt-files --verify --verify-files --list-keys --list-public-keys --list-signatures --list-sigs --check-signatures --check-sigs --fingerprint --list-secret-keys --generate-key --gen-key --quick-generate-key --quick-gen-key --quick-add-uid --quick-adduid --quick-add-key --quick-addkey --quick-revoke-uid --quick-revuid --quick-set-expire --quick-set-primary-uid --full-generate-key --full-gen-key --generate-revocation --gen-revoke --delete-keys --delete-secret-keys --quick-sign-key --quick-lsign-key --quick-revoke-sig --sign-key --lsign-key --edit-key --key-edit --change-passphrase --passwd --generate-designated-revocation --desig-revoke --export --send-keys --receive-keys --recv-keys --search-keys --refresh-keys --locate-keys --locate-external-keys --fetch-keys --show-keys --export-secret-keys --export-secret-subkeys --export-ssh-key --import --fast-import --list-config --list-gcrypt-config --gpgconf-list --gpgconf-test --list-packets --export-ownertrust --import-ownertrust --update-trustdb --check-trustdb --fix-trustdb --list-trustdb --dearmor --dearmour --enarmor --enarmour --print-md --print-mds --gen-prime --gen-random --server --tofu-policy --delete-secret-and-public-keys --rebuild-keydb-caches --list-key --list-sig --check-sig --show-key --Monitor --verbose --no-verbose --quiet --no-tty --no-greeting --debug --debug-level --debug-all --debug-iolbf --display-charset --charset --options --no-options --logger-fd --log-file --logger-file --debug-quick-random --Configuration --homedir --faked-system-time --default-key --encrypt-to --no-encrypt-to --hidden-encrypt-to --encrypt-to-default-key --default-recipient --default-recipient-self --no-default-recipient --group --ungroup --no-groups --compliance --gnupg --no-pgp2 --no-pgp6 --no-pgp7 --no-pgp8 --rfc2440 --rfc4880 --rfc4880bis --openpgp --pgp6 --pgp7 --pgp8 --default-new-key-algo --min-rsa-length --always-trust --trust-model --photo-viewer --known-notation --agent-program --dirmngr-program --exit-on-status-write-error --limit-card-insert-tries --enable-progress-filter --temp-directory --exec-path --expert --no-expert --no-secmem-warning --require-secmem --no-require-secmem --no-permission-warning --dry-run --interactive --default-sig-expire --ask-sig-expire --no-ask-sig-expire --default-cert-expire --ask-cert-expire --no-ask-cert-expire --default-cert-level --min-cert-level --ask-cert-level --no-ask-cert-level --only-sign-text-ids --enable-large-rsa --disable-large-rsa --enable-dsa2 --disable-dsa2 --personal-cipher-preferences --personal-digest-preferences --personal-compress-preferences --default-preference-list --default-keyserver-url --no-expensive-trust-checks --allow-non-selfsigned-uid --no-allow-non-selfsigned-uid --allow-freeform-uid --no-allow-freeform-uid --preserve-permissions --default-cert-check-level --tofu-default-policy --lock-once --lock-multiple --lock-never --compress-algo --compression-algo --bzip2-decompress-lowmem --completes-needed --marginals-needed --max-cert-depth --trustdb-name --auto-check-trustdb --no-auto-check-trustdb --force-ownertrust --Input --multifile --input-size-hint --utf8-strings --no-utf8-strings --set-filesize --no-literal --set-notation --sig-notation --cert-notation --set-policy-url --sig-policy-url --cert-policy-url --sig-keyserver-url --Output --armor --armour --no-armor --no-armour --output --max-output --comment --default-comment --no-comments --emit-version --no-emit-version --no-version --not-dash-escaped --escape-from-lines --no-escape-from-lines --mimemode --textmode --no-textmode --set-filename --for-your-eyes-only --no-for-your-eyes-only --show-notation --no-show-notation --show-session-key --use-embedded-filename --no-use-embedded-filename --unwrap --mangle-dos-filenames --no-mangle-dos-filenames --no-symkey-cache --skip-verify --list-only --compress-level --bzip2-compress-level --disable-signer-uid --ImportExport --auto-key-locate --no-auto-key-locate --auto-key-import --no-auto-key-import --auto-key-retrieve --no-auto-key-retrieve --include-key-block --no-include-key-block --disable-dirmngr --keyserver --keyserver-options --key-origin --import-options --import-filter --export-options --export-filter --merge-only --allow-secret-key-import --Keylist --list-options --show-photos --no-show-photos --show-policy-url --no-show-policy-url --with-colons --with-tofu-info --with-key-data --with-sig-list --with-sig-check --with-fingerprint --with-subkey-fingerprint --with-subkey-fingerprints --with-icao-spelling --with-keygrip --with-secret --with-wkd-hash --with-key-origin --fast-list-mode --fixed-list-mode --legacy-list-mode --print-pka-records --print-dane-records --keyid-format --show-keyring --recipient --hidden-recipient --recipient-file --hidden-recipient-file --remote-user --throw-keyids --no-throw-keyids --local-user --trusted-key --sender --try-secret-key --try-all-secrets --no-default-keyring --no-keyring --keyring --primary-keyring --secret-keyring --skip-hidden-recipients --no-skip-hidden-recipients --override-session-key --override-session-key-fd --Security --s2k-mode --s2k-digest-algo --s2k-cipher-algo --s2k-count --require-backsigs --require-cross-certification --no-require-backsigs --no-require-cross-certification --verify-options --enable-special-filenames --no-random-seed-file --no-sig-cache --ignore-time-conflict --ignore-valid-from --ignore-crc-error --ignore-mdc-error --disable-cipher-algo --disable-pubkey-algo --cipher-algo --digest-algo --cert-digest-algo --override-compliance-check --allow-weak-key-signatures --allow-weak-digest-algos --weak-digest --allow-multisig-verification --allow-multiple-messages --no-allow-multiple-messages --batch --no-batch --yes --no --status-fd --status-file --attribute-fd --attribute-file --command-fd --command-file --passphrase --passphrase-fd --passphrase-file --passphrase-repeat --pinentry-mode --force-sign-key --request-origin --display --ttyname --ttytype --lc-ctype --lc-messages --xauthority --no-autostart --forbid-gen-key --require-compliance --use-only-openpgp-card --rfc2440-text --no-rfc2440-text --personal-cipher-prefs --personal-digest-prefs --personal-compress-prefs --sign-with --user --use-agent --no-use-agent --gpg-agent-info --reader-port --ctapi-driver --pcsc-driver --disable-ccid --honor-http-proxy --tofu-db-format --strict --no-strict --load-extension --sk-comments --no-sk-comments --compress-keys --compress-sigs --force-v3-sigs --no-force-v3-sigs --force-v4-certs --no-force-v4-certs --no-mdc-warning --force-mdc --no-force-mdc --disable-mdc --no-disable-mdc --x-sequoia-parcimonie --x-sequoia-autostart-parcimonie --x-sequoia-parcimonie-daemonize --version --warranty --dump-option-table --dump-options --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --tofu-policy)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --options)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --log-file)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --default-key)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --encrypt-to)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --group)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --output)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -z)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --auto-key-locate)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --recipient)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -r)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --local-user)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -u)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

if [[ "${BASH_VERSINFO[0]}" -eq 4 && "${BASH_VERSINFO[1]}" -ge 4 || "${BASH_VERSINFO[0]}" -gt 4 ]]; then
    complete -F _gpg-sq -o nosort -o bashdefault -o default gpg-sq
else
    complete -F _gpg-sq -o bashdefault -o default gpg-sq
fi
